src/FSharp.Core/fslib-extra-pervasives.fsi

```diff
@@ -509,6 +509,15 @@ namespace Microsoft.FSharp.Core.CompilerServices
         /// <returns></returns>
         abstract ResolveTypeName : typeName: string -> Type
 
+    [<RequireQualifiedAccess; StructuralEquality; StructuralComparison>]
+    type TypeProviderDiagnosticSeverity =
+        | Warning
+        | Error
+
+    [<AllowNullLiteral>]
+    type ITypeProviderDiagnosticsContext =
+        abstract ReportDiagnostic: staticParameterName: string * rangeInParameterIfString: (int * int) option * message: string * severity: TypeProviderDiagnosticSeverity -> unit
```

Comment:
I think it might be a good idea to have `ReportDiagnostics`, allowing the TP to report multiple errors at once before the control is taken away. For example, multiple problems in a query string could be highlighted at the same time, not one by one as they are fixed.